### ISSUE TITLE
Prevent common accidental type mismatches

### ISSUE THREAD

**Issue by ZevEisenberg** [Raizlabs/Anchorage#13]
```swift
viewA.horizontalAnchors == viewB.edgeAnchors
viewA.horizontalAnchors == viewB.verticalAnchors
```

These both compile, but then crash at runtime because they are invalid constraints. The types exposed by Anchorage should turn these cases into compile errors.

**Comment by KingOfBrian** [comment]
I wonder if Phantom types would be helpful here -- https://www.natashatherobot.com/swift-money-phantom-types/

But I'm not sure it will. I think I would remove EdgeAnchors and its multi-use behavior with 3 structs for Edges, Horizontal and Vertical. There may be a protocol that can help with re-use.

**Comment by KingOfBrian** [comment]
Fixed in 3.0
